seqr/views/apis/report_api.py

```diff
@@ -893,6 +932,112 @@ def _get_experiment_ids(airtable_sample, airtable_metadata):
         'aligned_dna_short_read_id': f'{experiment_dna_short_read_id}_1'
     }
 
+
+def _get_validated_gregor_files(file_data):
+    errors = []
+    warnings = []
+    try:
+        validators, required_tables = _load_data_model_validators()
+    except Exception as e:
+        warnings.append(f'Unable to load data model for validation: {e}')
+        validators = {}
+        required_tables = set()
+
+    missing_tables = required_tables.difference({f[0] for f in file_data})
+    if missing_tables:
+        warnings.append(
+            f'The following tables are required in the data model but absent from the reports: {", ".join(missing_tables)}'
+        )
+
+    files = []
+    for file_name, data in file_data:
+        columns = TABLE_COLUMNS[file_name]
+        files.append([file_name, columns, data])
+
+        table_validator = validators.get(file_name)
+        if not table_validator:
+            warnings.append(f'No data model found for "{file_name}" table so no validation was performed')
+            continue
+
+        extra_columns = set(columns).difference(table_validator.keys())
+        if extra_columns:
+            col_summary = ', '.join(sorted(extra_columns))
+            warnings.append(
+                f'The following columns are included in the "{file_name}" table but are missing from the data model: {col_summary}'
+            )
+        missing_columns = set(table_validator.keys()).difference(columns)
+        if missing_columns:
+            col_summary = ', '.join(sorted(missing_columns))
+            warnings.append(
+                f'The following columns are included in the "{file_name}" data model but are missing in the report: {col_summary}'
+            )
```

Comment:
We report warnings if the schema of the report seqr is generating differs from the one in the one in the data model, we only report errors when the data model and the report match up but the data in the report does not pass the data model validation .This line reports a wraning about a misalignment of the data model, and we report an error if any row is missing a required column below

Reply:
You may know that a missing column won't be validated in the column data validation. So no error will be reported for the missing column afterward.

Reply:
If the column is missing in the data model we obviously can't validate the data in that column it because there is no model to validate against. If a column is missing in the report we can't validate that column  because it doesn't exist. Therefore we warn the user about these cases. I don't understand what change you would want here, keeping in mind my above explanation about what we warn on